src/doom/p_inter.c

```diff
@@ -86,10 +86,12 @@ P_GiveAmmo
 	num = clipammo[ammo]/2;
     
     if (gameskill == sk_baby
-	|| gameskill == sk_nightmare)
+	|| gameskill == sk_nightmare
+    || (crispy->moreammo && !demoplayback))
```

Comment:
`critical->moreammo` please without the extra check.

Reply:
Is that not going to desync the built-in demos?

Reply:
The `critical` struct takes care of that. It is the same as the `crispy` but all zero durin demo and network sessions. But you are probably right, it is a lot easier to understand if you leave it as it is.